### PR TITLE
Adds Beaker for acceptance testing of Boxen

### PR DESCRIPTION
**Pull request by petems**
I need to clean up the setup a bit more

This adds support to run acceptance tests using [Beaker](http://github.com/puppetlabs/beaker). This uses an OSX Vagrant box, runs the standard setup to get it working, then runs Boxen as the 'vagrant' user :+1: 

@dgoodlad any thoughts?

I'll add a README.md for the acceptance test too to say how to setup and such :smile:

### CODE REVIEW

File: spec/Gemfile.beaker [hunk 1]
```diff
@@ -0,0 +1,8 @@
+source "https://rubygems.org"
+
+group :system_tests do
+  gem 'beaker', :github => 'petems/beaker', :branch => 'MAINT-fix_mkdir_bug', :require => false
```
Comment (seanknox) [inline spec/Gemfile.beaker]: Can we use the official beaker repo instead of your fork?

Comment (petems) [inline spec/Gemfile.beaker]: Yeah, this is just because there was a bug on master, but I think its fixed by now, I'll change this back

Comment (petems) [inline spec/Gemfile.beaker]: Fixed :+1:

**Comment by petems** [comment]
Added README.md with basic instructions. 

There's probably a smarter way of handling the seperate Gemfile, any suggestions welcome :+1:

**Comment by radeksimko** [comment]
FYI: I was facing some issues with keychain access timing out during long-running operations, so I did some changes to that script:
https://github.com/radeksimko/vagrant-osx/blob/master/provision/setup-keychain.sh#L8
Feel free to update yours.

**Comment by jacobbednarz** [comment]
Closed in favour of #835. Please open another PR using serverspec if you'd like to add more coverage.

**Pull request closed without merging**